### ISSUE TITLE
relation widget allows selection of unpublished content across websites

### ISSUE THREAD

**Issue by gumaerc** [mitodl/ocw-studio#520]
### Steps to Reproduce

 - Define a relation widget in a starter that references content in another website
 - Create some content in the related website's field and save it but don't publish it
 - Create a site with the starter using the relation widget, and click the select field to bring up the available options

### Expected Behavior

Unpublished content is not selectable

### Actual Behavior

Unpublished content is selectable

**Comment by pdpinch** [comment]
@mbertrand do we have a way to know if a particular resource in a website has been published? 

See @noisecapella's comment for context: https://github.com/mitodl/ocw-studio/pull/525#issuecomment-925804312

**Comment by mbertrand** [comment]
No, when published the resource files are merged from the main branch on git to the appropriate preview/release branch but that is not tracked in the app except at the website level.  Would probably need to add another couple of fields to `WebsiteContent` (`published_draft`, `published_live`, could be booleans or dates) that would be updated either after the git merge or after the concourse pipeline run.

**Comment by mbertrand** [comment]
Maybe one way to tell without adding additional fields is comparison of the resource's `created_on` date with the website's `publish_date`.  If it was created before the website's `publish_date` then it should have been published.

**Comment by pdpinch** [comment]
I think @alicewriteswrongs solved this somehow for course lists. Do we need to implement the same thing here?

**Comment by alicewriteswrongs** [comment]
yes for course collections I was able to add an optional filter to the listing API so that we can fetch only published courses, then in the UI we use that API option when the courses are being selected.

**Comment by alicewriteswrongs** [comment]
I think we could implement something analogous for course content along the lines of what @mbertrand suggested above
